### ISSUE TITLE
Connection failure

### ISSUE THREAD

**Issue by 74cmonty** [nextcloud/android#3101]
### Actual behaviour
I have completed Nextcloud 14 installation on server running Alpine Linux 3.8.
I can login via WebUI w/o issues as admin.

I completed setup with creation of several user accounts incl. 2-factor-authentication.
I validated if this is working with WebUI again and there are no issues.

However, when I try to connect to NC14 using Android Client version 3.3.0 I get an error: Internal Server Failure (see [here](https://ibb.co/cLWa6p))

After a while I get an option to "Return to old registration method" (see [here](https://ibb.co/ikMKXU)).
The new login UI in the App then looks like [this](https://ibb.co/nGKqp9).

### Steps to reproduce
1. Start Android Client
2. Establish connection to NC14 server

### Environment data
Android version:
8.1.0
Device model: 
Sony Xperia Z3
Stock or customized system:
Omni-Rom 8.1.0-20180616-z3-nailyk
Nextcloud app version:
3.3
Nextcloud server version:
14.0.1
#### Web server error log
```
Stack trace:
#0 /usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(125): OC\AppFramework\Utility\SimpleContainer->resolve('defaultTokenPro...')
#1 /usr/share/webapps/nextcloud/lib/private/ServerContainer.php(132): OC\AppFramework\Utility\SimpleContainer->query('defaultTokenPro...')
#2 /usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(81): OC\ServerContainer->query('defaultTokenPro...')
#3 /usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(104): OC\AppFramework\Utility\SimpleContainer->buildClass(Object(ReflectionClass))
#4 /usr/share/webapps/nextcloud/lib/private/AppFramework/Utility/SimpleContainer.php(125): OC\
2018/09/28 12:10:51 [warn] 932#932: *3105 an upstream response is buffered to a temporary file /var/tmp/nginx/fastcgi/1/00/0000000001 while reading upstream, client: 10.0.0.2, server: <fqhn_nc_server>, request: "GET /remote.php/webdav/Nextcloud%20Manual.pdf HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "<fqhn_nc_server>"
```

#### Nextcloud log (data/nextcloud.log)
```
{"reqId":"xyBjC6rhxpC5MtiJZKeI","level":3,"time":"2018-09-28T10:00:05+00:00","remoteAddr":"10.0.0.2","user":"--","app":"index","method":"GET","url":"\/index.php\/login\/flow","message":{"Exception":"OCP\\AppFramework\\QueryException","Message":"Could not resolve OC\\Core\\Controller\\ClientFlowLoginController! Class OCA\\OAuth2\\Db\\ClientMapper does not exist","Code":0,"Trace":[{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","line":125,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OC\\Core\\Controller\\ClientFlowLoginController"]},{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/AppFramework\/DependencyInjection\/DIContainer.php","line":451,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OC\\Core\\Controller\\ClientFlowLoginController"]},{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/AppFramework\/DependencyInjection\/DIContainer.php","line":424,"function":"queryNoFallback","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OC\\Core\\Controller\\ClientFlowLoginController"]},{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/AppFramework\/App.php","line":105,"function":"query","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OC\\Core\\Controller\\ClientFlowLoginController"]},{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OC\\Core\\Controller\\ClientFlowLoginController","showAuthPickerPage",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"core.ClientFlowLogin.showAuthPickerPage"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"core.ClientFlowLogin.showAuthPickerPage"}]},{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/Route\/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"core.ClientFlowLogin.showAuthPickerPage"}]},{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/base.php","line":987,"function":"match","class":"OC\\Route\\Router","type":"->","args":["\/login\/flow"]},{"file":"\/usr\/share\/webapps\/nextcloud\/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","Line":110,"CustomMessage":"--"},"userAgent":"Sony Xperia Z3","version":"14.0.1.1"}
```

**Comment by tobiasKaminsky** [comment]
@74cmonty this sounds like a server configuration issue.

Can you create us a test account and send the credentials to tobias at nextcloud dot com with a reference to this issue?

**Comment by 74cmonty** [comment]
Hi Tobias,
how can I send you an encrypted email with your user credentials for my Nextcloud server?
I searched for your public key on hkps://hkps.pool.sks-keyservers.net to use PGP email encyption w/o success.

Please advice.

**Comment by 74cmonty** [comment]
**Update:**
- I can reproduce this issue with other Android devices using the same app from F-Droid.
- Using the old authorization method works for this app and I can use it.

**Comment by 74cmonty** [comment]
I have rebuild the server (same release 14.0.1) and the error still occurs:
```
{"reqId":"HyH7F0doQPEDdTFeFwwf","level":3,"time":"2018-10-12T15:15:01+00:00","remoteAddr":"10.0.0.100","user":"--","app":"index","method":"GET","url":"\/index.php\/login\/flow","message":{"Exception":"OCP\\AppFramework\\QueryException","Message":"Could not resolve OC\\Core\\Controller\\ClientFlowLoginController! Class OCA\\OAuth2\\Db\\ClientMapper does not exist","Code":0,"Trace":[{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","line":125,"function":"resolve","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OC\\Core\\Controller\\ClientFlowLoginController"]},{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/AppFramework\/DependencyInjection\/DIContainer.php","line":451,"function":"query","class":"OC\\AppFramework\\Utility\\SimpleContainer","type":"->","args":["OC\\Core\\Controller\\ClientFlowLoginController"]},{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/AppFramework\/DependencyInjection\/DIContainer.php","line":424,"function":"queryNoFallback","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OC\\Core\\Controller\\ClientFlowLoginController"]},{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/AppFramework\/App.php","line":105,"function":"query","class":"OC\\AppFramework\\DependencyInjection\\DIContainer","type":"->","args":["OC\\Core\\Controller\\ClientFlowLoginController"]},{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/AppFramework\/Routing\/RouteActionHandler.php","line":47,"function":"main","class":"OC\\AppFramework\\App","type":"::","args":["OC\\Core\\Controller\\ClientFlowLoginController","showAuthPickerPage",{"__class__":"OC\\AppFramework\\DependencyInjection\\DIContainer"},{"_route":"core.ClientFlowLogin.showAuthPickerPage"}]},{"function":"__invoke","class":"OC\\AppFramework\\Routing\\RouteActionHandler","type":"->","args":[{"_route":"core.ClientFlowLogin.showAuthPickerPage"}]},{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/Route\/Router.php","line":297,"function":"call_user_func","args":[{"__class__":"OC\\AppFramework\\Routing\\RouteActionHandler"},{"_route":"core.ClientFlowLogin.showAuthPickerPage"}]},{"file":"\/usr\/share\/webapps\/nextcloud\/lib\/base.php","line":987,"function":"match","class":"OC\\Route\\Router","type":"->","args":["\/login\/flow"]},{"file":"\/usr\/share\/webapps\/nextcloud\/index.php","line":42,"function":"handleRequest","class":"OC","type":"::","args":[]}],"File":"\/usr\/share\/webapps\/nextcloud\/lib\/private\/AppFramework\/Utility\/SimpleContainer.php","Line":110,"CustomMessage":"--"},"userAgent":"Sony Xperia Z3","version":"14.0.1.1"}
```

**Comment by 74cmonty** [comment]
fixed with NC 14.0.3 (installed from source)
